### PR TITLE
Lps 135570 1

### PR DESCRIPTION
**Pull request by tinatian**

### CODE REVIEW

File: modules/core/portal-web/src/main/java/com/liferay/portal/web/internal/session/replication/SessionReplicationHttpSessionWrapper.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		ByteBuffer byteBuffer = serializer.toByteBuffer();
+
+		super.setAttribute(
+			_SERIALIZED_ATTRIBUTE_PREFIX.concat(name), byteBuffer.array());
+	}
+
+	private boolean _isSafeClass(Class<?> clazz) {
+		ClassLoader classLoader = clazz.getClassLoader();
+
+		if ((classLoader == String.class.getClassLoader()) ||
```
Comment (tinatian) [inline modules/core/portal-web/src/main/java/com/liferay/portal/web/internal/session/replication/SessionReplicationHttpSessionWrapper.java:83]: @dantewang Could you please check if the classloader for String.class and HttpSession.class same?

**Comment by tinatian** [comment]
@dantewang Please also implement the getAttributeNames and removeAttribute for SessionReplicationHttpSessionWrapper

File: modules/core/portal-web/src/main/java/com/liferay/portal/web/internal/session/replication/SessionReplicationHttpSessionWrapper.java (hunk 1 above)
Comment (dantewang) [reply modules/core/portal-web/src/main/java/com/liferay/portal/web/internal/session/replication/SessionReplicationHttpSessionWrapper.java:83]: @tinatian I did check `HttpSession`'s class loader, and it was a `URLClassLoader` with jars in `tomcat-x.x.x/lib` & `tomcat-x.x.x/bin` as class path. I think `java.lang.String` should be loaded by JVM's system/bootstrap class loader, I'll double check.

Comment (dantewang) [reply modules/core/portal-web/src/main/java/com/liferay/portal/web/internal/session/replication/SessionReplicationHttpSessionWrapper.java:83]: @tinatian I can confirm that `String`'s class loader is `null`, which is the bootstrap class loader for JDK8.

**Comment by dantewang** [comment]
See https://github.com/tinatian/liferay-portal/pull/2345

**Pull request closed without merging**